tests/components/prometheus/test_init.py

```diff
@@ -183,6 +411,24 @@ async def test_view_empty_namespace(hass, hass_client):
         'friendly_name="Ecobee"} 24.0' in body
     )
 
+
+async def test_humidifier(hass, hass_client):
+    """Test prometheus metrics for battery."""
+    assert await async_setup_component(
+        hass,
+        "conversation",
```

Comment:
Why do we set up conversation?

Reply:
The `demo` platform needs  `conversation` as a dependency and loads it.
`prometheus` and `conversation` register resources to the `http` router.
If I setup the `demo` platform after `prometheus` following `RuntimeError` is thrown: `Cannot register a resource into frozen router`.

Is there a better way or should `await hass.async_block_till_done()` not be called after `prometheus` is setup?


## Log
```
DEBUG    homeassistant.core:core.py:721 Bus:Handling <Event service_registered[L]: domain=conversation, service=process>
ERROR    homeassistant.setup:setup.py:264 Error during setup of component conversation
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/setup.py", line 254, in _async_setup_component
    result = await task
  File "/workspaces/core/homeassistant/components/conversation/__init__.py", line 71, in async_setup
    hass.http.register_view(ConversationProcessView())
  File "/workspaces/core/homeassistant/components/http/__init__.py", line 278, in register_view
    view.register(self.app, self.app.router)
  File "/workspaces/core/homeassistant/components/http/view.py", line 95, in register
    routes.append(router.add_route(method, url, handler))
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_urldispatcher.py", line 1100, in add_route
    resource = self.add_resource(path, name=name)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_urldispatcher.py", line 1085, in add_resource
    self.register_resource(resource)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_urldispatcher.py", line 1047, in register_resource
    raise RuntimeError("Cannot register a resource into frozen router.")
RuntimeError: Cannot register a resource into frozen router.
DEBUG    homeassistant.core:core.py:721 Bus:Handling <Event state_changed[L]: entity_id=persistent_notification.invalid_config, old_state=None, new_state=<state persistent_notification.invalid_config=notifying; title=Invalid config, friendly_name=Invalid config, message=The following integrations and platforms could not be set up:

 - [conversation](https://www.home-assistant.io/integrations/conversation)
```

Reply:
In my new pull request #61355 to fix  #50382 entities have to be registered in the entity registry.

Is it ok to setup the integrations then or should I manually register the entity in the registry and set a state in the state machine?

Reply:
Just register an entity entry in the entity registry for the test as needed.

Reply:
You can do something like this:
```py
    entity_reg = er.async_get(hass)
    entry = entity_reg.async_get_or_create(DOMAIN, "test", "5678")
    hass.states.async_set(entry.entity_id, STATE_ALARM_PENDING)
```